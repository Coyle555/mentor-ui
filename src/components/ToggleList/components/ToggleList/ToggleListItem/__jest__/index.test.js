jest.mock('react-spring/renderprops', () => {
	return { Spring: props => (
		<div>
			<div>{JSON.stringify(props)}</div>
			{props.children({})}
		</div>
	)};
});

import React from 'react';
import renderer from 'react-test-renderer';
import { cleanup, fireEvent, render } from '@testing-library/react';
import { ToggleListItem } from '../index';

afterEach(cleanup);

test('Default toggle list item', () => {
	const tree = renderer.create(<ToggleListItem content="Foo content" title="Foo" />).toJSON();

	expect(tree).toMatchSnapshot();
});

test('Opening content on a toggle list item', () => {
	const { getByText } = render(<ToggleListItem content="content" title="Foo" />);

	fireEvent.click(getByText('Foo'));
	expect(getByText('content')).toBeTruthy();
});
