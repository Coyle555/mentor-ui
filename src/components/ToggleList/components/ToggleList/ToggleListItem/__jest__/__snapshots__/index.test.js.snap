// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default toggle list item 1`] = `
<div
  className="mui-toggle-list-item"
>
  <div
    className="title"
    onClick={[Function]}
  >
    <div>
      <div>
        {"from":{"transform":"rotate(0deg)"},"to":{"transform":"rotate(0deg)"}}
      </div>
      <i
        className="fas fa-caret-right icon"
        style={Object {}}
      />
    </div>
    Foo
  </div>
</div>
`;

exports[`Expanded toggle list item 1`] = `
<div
  className="mui-toggle-list-item"
>
  <div
    className="mui-toggle-list-item-title"
    onClick={[Function]}
  >
    <div>
      <div>
        {"from":{"transform":"rotate(0deg)"},"to":{"transform":"rotate(90deg)"}}
      </div>
      <i
        className="fas fa-caret-right icon"
        style={Object {}}
      />
    </div>
    Foo
  </div>
  <div>
    <div>
      {"from":{"opacity":0},"to":{"opacity":1}}
    </div>
    <div
      className=""
      style={Object {}}
    >
      Foo content
    </div>
  </div>
</div>
`;

exports[`Toggle list item with custom class for the title 1`] = `
<div
  className="mui-toggle-list-item"
>
  <div
    className="title custom-class"
    onClick={[Function]}
  >
    <div>
      <div>
        {"from":{"transform":"rotate(0deg)"},"to":{"transform":"rotate(0deg)"}}
      </div>
      <i
        className="fas fa-caret-right icon"
        style={Object {}}
      />
    </div>
    Foo
  </div>
</div>
`;
